Author: Julio Merino <jmmv@google.com>

Properly configure SQLite on Linux and OS X

Add all necessary -DHAVE_* flags that result from a run of the configure
script on Linux and OS X so that we build SQLite properly.  In
particular, this fixes mattn/go-sqlite3 #285 in which I spotted erratic
backoff behavior due to the lack of usleep support.

--- deps/src/github.com/mattn/go-sqlite3/sqlite3.go
+++ deps/src/github.com/mattn/go-sqlite3/sqlite3.go
@@ -12,6 +12,47 @@
 #cgo CFLAGS: -DSQLITE_TRACE_SIZE_LIMIT=15
 #cgo CFLAGS: -DSQLITE_DISABLE_INTRINSIC
 #cgo CFLAGS: -Wno-deprecated-declarations
+
+// Darwin configuration taken from a configure run of SQLite 3.11.0 on OS X El Capitan.
+#cgo darwin CFLAGS: -DHAVE_DECL_STRERROR_R=1
+#cgo darwin CFLAGS: -DHAVE_DLFCN_H=1
+#cgo darwin CFLAGS: -DHAVE_EDITLINE=1
+#cgo darwin CFLAGS: -DHAVE_FDATASYNC=1
+#cgo darwin CFLAGS: -DHAVE_GMTIME_R=1
+#cgo darwin CFLAGS: -DHAVE_INTTYPES_H=1
+#cgo darwin CFLAGS: -DHAVE_LOCALTIME_R=1
+#cgo darwin CFLAGS: -DHAVE_MEMORY_H=1
+#cgo darwin CFLAGS: -DHAVE_STDINT_H=1
+#cgo darwin CFLAGS: -DHAVE_STDLIB_H=1
+#cgo darwin CFLAGS: -DHAVE_STRERROR_R=1
+#cgo darwin CFLAGS: -DHAVE_STRINGS_H=1
+#cgo darwin CFLAGS: -DHAVE_STRING_H=1
+#cgo darwin CFLAGS: -DHAVE_SYS_STAT_H=1
+#cgo darwin CFLAGS: -DHAVE_SYS_TYPES_H=1
+#cgo darwin CFLAGS: -DHAVE_UNISTD_H=1
+#cgo darwin CFLAGS: -DHAVE_USLEEP=1
+#cgo darwin CFLAGS: -DSTDC_HEADERS=1
+
+// Linux configuration taken from a configure run of SQLite 3.11.0 on Ubuntu 14.04.4 LTS.
+#cgo linux CFLAGS: -DHAVE_DECL_STRERROR_R=1
+#cgo linux CFLAGS: -DHAVE_DLFCN_H=1
+#cgo linux CFLAGS: -DHAVE_FDATASYNC=1
+#cgo linux CFLAGS: -DHAVE_GMTIME_R=1
+#cgo linux CFLAGS: -DHAVE_INTTYPES_H=1
+#cgo linux CFLAGS: -DHAVE_LOCALTIME_R=1
+#cgo linux CFLAGS: -DHAVE_MEMORY_H=1
+#cgo linux CFLAGS: -DHAVE_POSIX_FALLOCATE=1
+#cgo linux CFLAGS: -DHAVE_STDINT_H=1
+#cgo linux CFLAGS: -DHAVE_STDLIB_H=1
+#cgo linux CFLAGS: -DHAVE_STRERROR_R=1
+#cgo linux CFLAGS: -DHAVE_STRINGS_H=1
+#cgo linux CFLAGS: -DHAVE_STRING_H=1
+#cgo linux CFLAGS: -DHAVE_SYS_STAT_H=1
+#cgo linux CFLAGS: -DHAVE_SYS_TYPES_H=1
+#cgo linux CFLAGS: -DHAVE_UNISTD_H=1
+#cgo linux CFLAGS: -DHAVE_USLEEP=1
+#cgo linux CFLAGS: -DSTDC_HEADERS=1
+
 #ifndef USE_LIBSQLITE3
 #include <sqlite3-binding.h>
 #else
